### PR TITLE
Fix Interacting with Air Alarm and Fire Alarm.

### PR DESCRIPTION
**Pull request by Charlese2**
## About the PR 
There is an issue with how rotated sprite offsets are handled that causes interaction issues with Air Alarms and Fire Alarms.
This works around using offsets to avoid the issue.
It might mess with already mapped Alarms as it is actually in the wall without the offset.
**Screenshots**

Location Before
image
Location After
image

**Changelog**

:cl:
- fix: Fixed some of the interaction issues with Air Alarms and Fire Alarms.

### CODE REVIEW

File: Content.Server/Atmos/Monitor/Systems/AirAlarmSystem.cs [hunk 1]
```diff
@@ -188,7 +188,7 @@ private void OnClose(EntityUid uid, AirAlarmComponent component, BoundUIClosedEv
 
         private void OnInteract(EntityUid uid, AirAlarmComponent component, InteractHandEvent args)
         {
-            if (!_interactionSystem.InRangeUnobstructed(args.User, args.Target))
+            if (!_interactionSystem.InRangeUnobstructed(args.User, args.Target, 2, Shared.Physics.CollisionGroup.Impassable, null, true))
```
Comment (ElectroJr) [inline Content.Server/Atmos/Monitor/Systems/AirAlarmSystem.cs:191]: If all you want to do is specify the ignoreInsideBlocker, you don't have to override all the other defaults. you can (and should) just do:
```suggestion
            if (!_interactionSystem.InRangeUnobstructed(args.User, args.Target, ignoreInsideBlocker: true))
```
and similarly in fire alarm system

File: Content.Server/Atmos/Monitor/Systems/AtmosMonitoringSystem.cs [hunk 2]
```diff
@@ -119,32 +118,23 @@ private void OnAtmosMonitorShutdown(EntityUid uid, AtmosMonitorComponent compone
...
-
                 var rotPos = transform.LocalRotation.RotateVec(new Vector2(0, -1));
-                transform.Anchored = false;
                 coords = coords.Offset(rotPos);
-                transform.Coordinates = coords;
-
-                appearance.SetData("offset", -rotPos);
-
-                transform.Anchored = true;
             }
```
Comment (metalgearsloth) [inline Content.Server/Atmos/Monitor/Systems/AtmosMonitoringSystem.cs:138]: I don't think this works does it?

File: Content.Server/Atmos/Monitor/Systems/AirAlarmSystem.cs (hunk 1 above)
Comment (Charlese2) [reply Content.Server/Atmos/Monitor/Systems/AirAlarmSystem.cs:191]: Good to know.

File: Content.Server/Atmos/Monitor/Systems/AtmosMonitoringSystem.cs (hunk 2 above)
Comment (Charlese2) [reply Content.Server/Atmos/Monitor/Systems/AtmosMonitoringSystem.cs:138]: It does. There is probably a better way though.
It gets the coords of were it was going to place the entity (in front of it) and uses that to get the Air (GasMixture).

File: Content.Server/Atmos/Monitor/Systems/AtmosMonitoringSystem.cs [hunk 3]
```diff
@@ -119,32 +118,23 @@ private void OnAtmosMonitorShutdown(EntityUid uid, AtmosMonitorComponent compone
...
             var coords = transform.Coordinates;
 
             if (_atmosphereSystem.IsTileAirBlocked(coords))
             {
-
                 var rotPos = transform.LocalRotation.RotateVec(new Vector2(0, -1));
-                transform.Anchored = false;
                 coords = coords.Offset(rotPos);
-                transform.Coordinates = coords;
-
```
[the remark this answers is not in the document]
Comment (ElectroJr) [reply Content.Server/Atmos/Monitor/Systems/AtmosMonitoringSystem.cs:144]: Though I have to say, I hate the old/current solution of an entity using using appearance offsets to pretend to be on a tile that they're not actually on.

[the remark this answers is not in the document]
Comment (Charlese2) [reply Content.Server/Atmos/Monitor/Systems/AtmosMonitoringSystem.cs:144]: I think it will cause issues.

**Comment by metalgearsloth** [comment]
Uhh I forgot but do we still need this @ElectroJr

**Comment by ElectroJr** [comment]
> Uhh I forgot but do we still need this @ElectroJr

Well currently the air alarms are in front of the wall instead of on them, but we decided during the maintainer meeting to just have things on walls. IIRC this PR basically moved them back onto the wall, which we want, but the issue was it then no longer supports the `TileFireEvent`.

So either:
- Keep as is (after checking everything works, I never actually tested in game), and just remove `TileFireEvent` functionality for now
- Add some sort of air-probe entity that connects to the air alarm. Either automatically & invisibly, or maybe something under floor tiles like the existing air vents/pipes.

**Comment by metalgearsloth** [comment]
@Zumorica any thoughts on the TileFireEvent?

**Comment by Zumorica** [comment]
> @Zumorica any thoughts on the TileFireEvent?

idk, use dummy entities or something?

**Review by metalgearsloth — APPROVED** [review]
Seems okay apart from the fire thing

**Comment by metalgearsloth** [comment]
Ping for re-open, still want this in-game.

**Pull request closed without merging**